File: _inc/laravel/app/Models/Abstracts/Entities/Deliverable.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\{Factories\HasFactory, Model};
use Illuminate\Support\Str;

abstract class Deliverable extends Model
{
  use HasFactory;
  protected string $keyType = 'string';
  public bool $incrementing = false;
  protected array $casts = [
    'billing_phone_verified_at' => 'datetime',
    'email_verified_at' => 'datetime',
    'is_active' => 'boolean',
    'shipping_phone_verified_at' => 'datetime',
  ];
  protected array $fillable = [
    'avatar',
    'avatar_url',
    'billing_address',
    'billing_city',
    'billing_country',
    'billing_phone',
    'billing_phone_verified_at',
    'billing_state',
    'billing_zip',
    'contact',
    'created_by',
    'email',
    'email_verified_at',
    'is_active',
    'lang',
    'name',
    'notes',
    'secondary_email',
    'shipping_address',
    'shipping_city',
    'shipping_country',
    'shipping_name',
    'shipping_phone',
    'shipping_phone_verified_at',
    'shipping_state',
    'shipping_zip',
  ];
  protected static function booted(): void
  {
    parent::booted();
    static::creating(function (self $model): void {
      $model->{$model->getKeyName()} = (string) Str::uuid();
    });
  }
}

File: _inc/_DEPRECATED_django/erp_prestech_backend/app/Models/_helpers/deliverable.py
from django.db import models
from django.core.files.storage import default_storage
from .default_timed import DefaultTimed
from .fields import (uuid_def_primary, default_char_field, default_user_creation,
                               default_text_field, VOID, TINY_BLANK_CHAR, FILE_SIZE_VALIDATOR, 
                               WORDED_NAME_VALIDATOR, PHONE_VALIDATOR, SHORT_BLANK_CHAR,
                               LANG_CHOICES, ZIP_VALIDATOR)
class Deliverable(DefaultTimed):
  
  class Meta:
    abstract = True
  
  id = models.UUIDField(**uuid_def_primary())
  lang = models.CharField(max_length=20, default='en', choices=LANG_CHOICES)
  name = default_char_field()
  email = models.EmailField(max_length=254, unique=True)
  secondary_email = models.EmailField(max_length=254, **VOID)
  contact = models.CharField(**TINY_BLANK_CHAR)
  avatar = models.ImageField(upload_to='avatars/', storage=default_storage, validators=[FILE_SIZE_VALIDATOR], **VOID)
  avatar_url = models.URLField(max_length=200, **VOID)
  email_verified_at = models.DateTimeField(**VOID)
  is_active = models.BooleanField(default=True)
  billing_country = default_char_field(validators=[WORDED_NAME_VALIDATOR])
  billing_state = default_char_field(validators=[WORDED_NAME_VALIDATOR])
  billing_city = default_char_field(validators=[WORDED_NAME_VALIDATOR])
  billing_phone = models.CharField(max_length=50, validators=[PHONE_VALIDATOR])
  billing_phone_verified_at = models.DateTimeField(**VOID)
  billing_zip = models.CharField(max_length=20, validators=[ZIP_VALIDATOR])
  billing_address = default_text_field(default='No billing address was given')
  shipping_name = default_char_field(voidable=True)
  shipping_country = models.CharField(**SHORT_BLANK_CHAR,validators=[WORDED_NAME_VALIDATOR])
  shipping_state = models.CharField(**SHORT_BLANK_CHAR,validators=[WORDED_NAME_VALIDATOR])
  shipping_city = models.CharField(**SHORT_BLANK_CHAR,validators=[WORDED_NAME_VALIDATOR])
  shipping_phone = models.CharField(**TINY_BLANK_CHAR,validators=[PHONE_VALIDATOR])
  shipping_phone_verified_at = models.DateTimeField(**VOID)
  shipping_zip = models.CharField(max_length=20, blank=True, validators=[ZIP_VALIDATOR])
  shipping_address = default_text_field(default='No shipping address was given')
  notes = default_text_field(length='medium', default='No notes were taken', **VOID)
  created_by = default_user_creation('%(class)s_created_by')